src/main/java/com/snowflake/kafka/connector/records/RecordService.java

```diff
@@ -287,14 +287,7 @@ private Map<String, Object> getMapFromJsonNodeForStreamingIngest(JsonNode node)
       String columnName = columnNames.next();
       JsonNode columnNode = node.get(columnName);
       Object columnValue;
-      if (columnNode.isArray()) {
```

Comment:
This is an interesting observation, I assume this was added for a reason, we will evaluate it to see if it will break anything, thanks!

Reply:
Looks like it was introduced in [PR](https://github.com/snowflakedb/snowflake-kafka-connector/pull/467/files#diff-7920dca33f32a43f73302b546c7877c26ceeeed3fd555a7dfb110d77b42658f7R251-R259) that enabled schematization and hasn't been updating since.

Reply:
If you serialize an entire array node, it's JSON elements won't be stringified. They will appear as "expected" in description. And that won't require to use `parse_json()`